File: src/main/java/random/telegramhomebot/model/HostStateDeserializer.java
package random.telegramhomebot.model;

import com.fasterxml.jackson.core.JsonParser;
import com.fasterxml.jackson.databind.DeserializationContext;
import com.fasterxml.jackson.databind.JsonDeserializer;
import com.fasterxml.jackson.databind.JsonNode;

import java.io.IOException;

public class HostStateDeserializer extends JsonDeserializer<HostState> {
	@Override
	public HostState deserialize(JsonParser jp, DeserializationContext ctxt) throws IOException {
		JsonNode node = jp.readValueAsTree();
		return HostState.valueOf(node.get(0).asText());
	}
}

File: src/main/java/random/telegramhomebot/services/scan/model/HostStateDeserializer.kt
package random.telegramhomebot.services.scan.model

import com.fasterxml.jackson.core.JsonParser
import com.fasterxml.jackson.databind.DeserializationContext
import com.fasterxml.jackson.databind.JsonDeserializer
import com.fasterxml.jackson.databind.JsonNode
import random.telegramhomebot.db.model.HostState
import java.io.IOException

class HostStateDeserializer : JsonDeserializer<HostState>() {
    @Throws(IOException::class)
    override fun deserialize(jp: JsonParser, ctxt: DeserializationContext): HostState {
        val node = jp.readValueAsTree<JsonNode>()
        return HostState.valueOf(node[0].asText())
    }
}